pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -38,6 +38,8 @@ def ones(*size, out=None, dtype=None, layout=None, device=None, requires_grad=Fa
 
 
 def full(size, fill_value, *, out=None, dtype=None, layout=None, device=None, requires_grad=False) -> Tensor:
+  if dtype is None:
+    dtype = type(fill_value).__name__ + "32"
```

Comment:
This is too hacky and wrong in a couple of cases (e.g. when `fill_value` is a `numpy.int32`).

Reply:
Also, it still does not follow the PyTorch logic.

Reply:
Yes, it also seemed a bit hacky to me. But we to just take `type(fill_value)` we would need to extend `_dtype` in `_C.py` and if we don't take care of the 32 this will also not work.

Reply:
The check should be:
```suggestion
    if isinstance(fill_value, int) or isinstance(fill_value, numpy.integer):
      dtype = "int64"
    else:
      dtype = "float32"  # actually torch.get_default_dtype()
```

Reply:
Do you already have `get_default_dtype`? Maybe add it and just return "float32" there for now.

The logic here might still be incomplete, e.g. for "bool".